src/transform/FooterContainer.js

```diff
@@ -0,0 +1,91 @@
+import './FooterContainer.css'
+import Polyfill from './Polyfill'
+
+/**
+ * Ensures the 'Read more' section header can always be scrolled to the top of the screen.
+ * @param {!Window} window
+ * @return {void}
+ */
+const updateBottomPaddingToAllowReadMoreToScrollToTop = window => {
+  const div = window.document.getElementById('pagelib_footer_container_ensure_can_scroll_to_top')
+  const currentPadding = parseInt(div.style.paddingBottom, 10) || 0
+  const height = div.clientHeight - currentPadding
+  const newPadding = Math.max(0, window.innerHeight - height)
+  div.style.paddingBottom = `${newPadding}px`
+}
+
+/**
+ * Allows native code to adjust footer container margins without having to worry about
+ * implementation details.
+ * @param {!number} margin
+ * @param {!Document} document
+ * @return {void}
+ */
+const updateLeftAndRightMargin = (margin, document) => {
+  const elements = Polyfill.querySelectorAll(document, `
+    #pagelib_footer_container_menu_heading, 
+    #pagelib_footer_container_readmore, 
+    #pagelib_footer_container_legal
+  `)
+  Array.from(elements)
+    .forEach(element => {
+      element.style.marginLeft = `${margin}px`
+      element.style.marginRight = `${margin}px`
+    })
+  const rightOrLeft = document.querySelector('html').dir === 'rtl' ? 'right' : 'left'
+  Array.from(Polyfill.querySelectorAll(document, '.pagelib_footer_menu_item'))
+    .forEach(element => {
+      element.style.backgroundPosition = `${rightOrLeft} ${margin}px center`
```

Comment:
Do these need to change dynamically or can these inline styles be replaced with a platform style sheet (no JavaScript needed)?

```css
.pagelib-platform-android .pagelib_footer_menu_item {
  padding-left: 4px;
  padding-right: 4px;
}

.pagelib-platform-ios .pagelib_footer_menu_item {
  padding-left: 10px;
  padding-right: 10px;
}

.pagelib-platform-android .content-rtl .pagelib_footer_menu_item {
  background-position: 4px center;
}

.pagelib-platform-ios .content-rtl .pagelib_footer_menu_item {
  background-position: 10px left;
}
```

The only JavaScript you would need is to examine the user agent and could be used by all future transforms. Something like:

```js file=PlatformTransform.js
const PLATFORM = { ANDROID: 'pagelib-platform-android', IOS:  'pagelib-platform-ios' }

const isAndroid = window => /android/i.test(window.navigator.userAgent)
const isIOs = window => /ipad|iphone|ipod/i.test(window.navigator.userAgent)

const classify = window => {
  const html = window.document.querySelector('html')
  if (isAndroid(window)) { html.classList.add(PLATFORM.ANDROID) }
  if (isIOs(window)) { html.classList.add(PLATFORM.IOS) }
}

export default { PLATFORM, classify }
```
I haven't tested this so it probably needs some tweaking to actually work. The regexs are from [mobfro](https://phabricator.wikimedia.org/diffusion/EMFR/browse/master/resources/mobile.startup/browser.js).